### ISSUE TITLE
Writable Measurement Typed As OTT_MEASURE

### ISSUE THREAD

**Issue by gr1pcoding** [zariiii9003/pycanape#8]
Thanks for the great library, super easy to use and got all the features I was interested in up and running pretty quickly.

The below if block is causing me trouble... My a2l has some writable measurement channels. I need to be able to write these via the Calibration class. When I comment out this if block, I’m able to get it working. On my setup for some reason writable measurements are showing as OTT_MEASURE, and not OTT_CALIBRATE like the API documentation says. 

`def _write_calibration_object_value(self, cov: TCalibrationObjectValue) -> None:
        if self.object_type != ObjectType.OTT_CALIBRATE:
            err_msg = "Cannot set value to a Measurement Object."
            raise TypeError(err_msg)`

I am not sure if others have this same problem or if it is unique to the CANape V16 version I'm running. In any case, commenting out the if block is my only remedy at the moment so looking for a more permanent fix. To see if it would hurt anything to remove the if block for all users, I tried calibrating some read-only measurements channels and it still errors out albeit with a different exception so in my mind it still works as intended: 

`File "C:\bin\tool_lib\site-packages\pycanape\cnp_api\cnp_prototype.py", line 2631, in _get_last_error
    raise CANapeError(error_code, error_msg, function.__name__)
pycanape.utils.CANapeError: Asap3WriteCalibrationObject failed (AEC_NOT_WRITE_ACCESS: Object has not write access)` 

Below is the a2l screenshot of the writable measurement that is showing up as OTT_MEASURE if helpful:

![image](https://github.com/user-attachments/assets/817e7140-9d40-40ed-aa33-fbe7b63180d2)

**Comment by zariiii9003** [comment]
Hi,  if the object is writable, then it is a calibration object and your a2l is incorrect. 
There is an ASAP2 editor in CANape, where you can convert your measurement object to a calibration object.

**Comment by gr1pcoding** [comment]
CANape has the option to make measurements read/write per the a2l screenshot. I don't have the capability to change them due to the auto-generation process merging a2l's from other sources outside of my control.

**Comment by zariiii9003** [comment]
I checked the ASAP2 spec, you're right. Writable Measurement objects exist.
![image](https://github.com/user-attachments/assets/2b7252bc-f31b-4564-9b0c-702dbe3b0c99)

But the CANape documentation says this:
![image](https://github.com/user-attachments/assets/fa903c8e-cdab-4561-bcb6-af90d5e52010)
So this seems like a CANape bug...

**Comment by gr1pcoding** [comment]
Yep I saw the same thing. Let me try it on a newer version of CANape first and see if it resolves. If not I can patch it myself if you're not interested in my proposed change of removing the if block

**Comment by zariiii9003** [comment]
> Yep I saw the same thing. Let me try it on a newer version of CANape first and see if it resolves. If not I can patch it myself if you're not interested in my proposed change of removing the if block

Sounds good, if newer versions still behave like this, we could replace that  `object_type` check with try-except 🤷‍♂️

**Comment by meganmlee** [comment]
I am facing the same issue where my writable measurements are showing as OTT_MEASURE and I have to comment those lines out for it to write to it.

@gr1pcoding did you find a fix when you used newer versions? I am currently using v18

It would be awesome if there was a better fix, otherwise I can patch it as well

**Comment by gr1pcoding** [comment]
@meganmlee Hi, yeah sorry I never really followed up on this and didn't have time to try other CANape versions. I still am working with a patch for the time being!

**Comment by meganmlee** [comment]
@gr1pcoding Gotcha, I'll just make a patch as well. thanks!

**Comment by zariiii9003** [comment]
I removed that object type check in v0.8.1, closing.
